### PR TITLE
[Improvement-13824][Resource Center] User can only visit resources under tenant defaultPath

### LINKED ISSUE #13824: [Improvement][Resource Center] Deny requests to visit other tenants' resources
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Currently, users can enter other tenant's directory and browse resources which do not belong to them. We can add check to deny such a request. 

image

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Radeity**
## Purpose of the pull request
- This PR closes https://github.com/apache/dolphinscheduler/issues/13824

## Brief change log
- Add new api `/resource/base-dir` for querying base resource dir which add constraint that breadcrumb navigation can only visit resources or files under tenant's resource paths.

## Verify this pull request
- Add some UT.

### CODE REVIEW

File: dolphinscheduler-ui/src/views/resource/index.tsx [hunk 1]
```diff
@@ -25,4 +27,15 @@ const resource = defineComponent({
   }
 })
 
+export function getBaseDir(resourceType: ResourceType) {
+  const baseDir = ref(String(""));
```
Comment (songjianet) [inline dolphinscheduler-ui/src/views/resource/index.tsx]: Please use single quotes and remove the semicolons.

File: dolphinscheduler-ui/src/views/resource/index.tsx [hunk 2]
```diff
@@ -15,7 +15,9 @@
  * limitations under the License.
  */
 
-import { defineComponent } from 'vue'
+import { defineComponent, ref } from 'vue'
+import { queryBaseDir } from "@/service/modules/resources";
```
Comment (songjianet) [inline dolphinscheduler-ui/src/views/resource/index.tsx]: Please remove the semicolon.

File: dolphinscheduler-ui/src/views/resource/index.tsx [hunk 3]
```diff
@@ -15,7 +15,9 @@
...
+import { ResourceType } from "@/views/resource/components/resource/types";
```
Comment (songjianet) [inline dolphinscheduler-ui/src/views/resource/index.tsx]: Please remove the semicolon.

File: dolphinscheduler-ui/src/views/resource/components/resource/index.tsx [hunk 4]
```diff
@@ -46,6 +46,7 @@ import styles from './index.module.scss'
 import type { Router } from 'vue-router'
 import Search from "@/components/input-search"
 import { ResourceType } from "@/views/resource/components/resource/types";
```
Comment (songjianet) [inline dolphinscheduler-ui/src/views/resource/components/resource/index.tsx]: Please remove the semicolon.

File: dolphinscheduler-ui/src/views/resource/components/resource/index.tsx [hunk 5]
```diff
@@ -46,6 +46,7 @@ import styles from './index.module.scss'
...
+import { getBaseDir } from "@/views/resource";
```
Comment (songjianet) [inline dolphinscheduler-ui/src/views/resource/components/resource/index.tsx]: Please remove the semicolon.

File: dolphinscheduler-ui/src/service/modules/resources/index.ts [hunk 6]
```diff
@@ -45,6 +45,16 @@ export function queryResourceListPaging(
   })
 }
 
+export function queryBaseDir(
+    params: ResourceTypeReq,
```
Comment (songjianet) [inline dolphinscheduler-ui/src/service/modules/resources/index.ts]: Please remove the comma.

**Review by songjianet — CHANGES_REQUESTED** [review]

**Comment by Radeity** [comment]
Hi, @songjianet , thanks for your comments, i've removed all semi-colons and modified the logic that only query `base-dir` after login and save it in `userStore`.

**Comment by songjianet** [comment]
> Hi, @songjianet , thanks for your comments, i've removed all semi-colons and modified the logic that only query `base-dir` after login and save it in `userStore`.

image

**Comment by Radeity** [comment]
> > Hi, @songjianet , thanks for your comments, i've removed all semi-colons and modified the logic that only query `base-dir` after login and save it in `userStore`.
> 
> image

Done.

**Review by songjianet — APPROVED** [review]

**Comment by Radeity** [comment]
Hi, @SbloodyS @caishunfeng , would you like to help review back-end part when you're available :D

**Review by SbloodyS — APPROVED** [review]
+1

**Comment by caishunfeng** [comment]
This pr fix the security problem, and I think we should cherry-pick to the next 3.1.x version, so I add the `3.1.x` label @zhongjiajie

**Comment by zhongjiajie** [comment]
> This pr fix the security problem, and I think we should cherry-pick to the next 3.1.x version, so I add the `3.1.x` label @zhongjiajie

cc @zhuangchong

**Comment by zhuangchong** [comment]
@Radeity The part code in the dev branch and the 3.1.9 branch has been greatly changed in the resource center. Can you submit a PR to the 3.1.9-prepare branch?

**Comment by zhongjiajie** [comment]
> @Radeity The part code in the dev branch and the 3.1.9 branch has been greatly changed in the resource center. Can you submit a PR to the 3.1.9-prepare branch?

Or maybe we can tell to security or reporter we could fix it in 3.2.1? @caishunfeng

**Comment by Radeity** [comment]
> @Radeity The part code in the dev branch and the 3.1.9 branch has been greatly changed in the resource center. Can you submit a PR to the 3.1.9-prepare branch?

Sorry for the late reply, currently, I'm busy with my thesis, do not have much time on this, can we consider @z
's opinion?

**Pull request merged**